### PR TITLE
Add documentation to metadata templates

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature
- Documentation

### Description

This PR adds the documentation for the new feature which is templates for metadata (labels and annotations).

### CODE REVIEW

File: documentation/book/proc-customizing-deployments.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+= Customizing deployments
+
+.Prerequisites
+
+* An {ProductPlatformName} cluster
+* A running Cluster Operator
+
+.Procedure
+
+. Edit the `template` property in the `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, or `KafkaMirrorMaker` resource.
```
Comment (stanlyDoge) [inline documentation/book/proc-customizing-deployments.adoc]: comma before `KafkaMirrorMaker`

File: documentation/book/assembly-customizing-deployments.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration.adoc
+
+[id='assembly-customizing-deployments-{context}']
+= Customizing deployments
+
+The {ProductPlatformName} resources (such as `Deployments`, `StatefulSets`, `Pods` or `Services`) created by {ProductName} can be always modified only by the operators managing them.
```
Comment (kyguy) [inline documentation/book/assembly-customizing-deployments.adoc]: can be always modified only -> can only be modified

File: documentation/book/assembly-customizing-deployments.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+When the user tries to change the manually, the operators will always revert them back to the original state.
+However, there are many situations when the user might want to modify them.
+For example, to add custom labels or annotations which control how are the `Pods` treated by other services such as Istio or how are the `Loadbalancer` type `Services` created by the cluster.
+{ProductName} allows to modify some of these aspects using the `template` properties in the {ProductName} custom resources.
```
Comment (kyguy) [inline documentation/book/assembly-customizing-deployments.adoc]: allows -> allows users

File: documentation/book/con-customizing-template-properties.adoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-template-properties-{context}']
+= Template properties
+
+The `template` property allows you to configure selected aspects of how are the resources created by {ProductName} configured.
```
Comment (kyguy) [inline documentation/book/con-customizing-template-properties.adoc]: how are the -> how
created - > are created
configured. -> .

File: documentation/book/con-customizing-template-properties.adoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+The `template` property can be used in following resources / properties:
```
Comment (kyguy) [inline documentation/book/con-customizing-template-properties.adoc]: following -> the following

File: documentation/book/con-customizing-template-properties.adoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+* Kafka.spec.kafka
+* Kafka.spec.zookeeper
+* Kafka.spec.entityOperator
+* KafkaConnect.spec
+* KafkaConnectS2I.spec
+* KafkaMirrorMakerSpec
+
+The following example shows the `template` being used to modify labels of Kafka broker `StatefulSet`:
```
Comment (kyguy) [inline documentation/book/con-customizing-template-properties.adoc]: Kafka -> a Kafka

File: documentation/book/proc-customizing-deployments.adoc [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+For example to modify the labels for the Kafka broker `StatefulSet`, use:
```
Comment (kyguy) [inline documentation/book/proc-customizing-deployments.adoc]: example -> example,

File: documentation/book/proc-customizing-deployments.adoc (hunk 1 above)
Comment (scholzj) [reply documentation/book/proc-customizing-deployments.adoc]: @stanlyDoge ??? It is already there.

Comment (stanlyDoge) [reply documentation/book/proc-customizing-deployments.adoc]: Oh, I thought it should not be there.

Comment (scholzj) [reply documentation/book/proc-customizing-deployments.adoc]: I think it should ... they are always adding it there when you have before it more than one thing. Let's see if @laidan6000 likes it.

File: documentation/book/con-customizing-labels-and-annotations.adoc [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            labels:
+                label1: value1
+                label2: value2
+            annotations:
+                annotation1: value1
+                annotation2: value2
+# ...
+----
+
+The `labels` and `annotations` fields can contain any labels or annotations which do not contain the string `strimzi.io`.
```
Comment (ppatierno) [inline documentation/book/con-customizing-labels-and-annotations.adoc]: "the reserved string"

File: documentation/book/assembly-customizing-deployments.adoc [hunk 9]
```diff
@@ -0,0 +1,18 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-deployment-configuration.adoc
+
+[id='assembly-customizing-deployments-{context}']
+= Customizing deployments
```
Comment (laidan6000) [inline documentation/book/assembly-customizing-deployments.adoc]: @ppatierno . Can you think of a more descriptive title for the assembly? Perhaps something like:

## Customizing deployments using templates

Suggestions welcome!

**Comment by scholzj** [comment]
What is it you don't like on the title?

File: documentation/book/con-customizing-template-properties.adoc [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+          labels:
+            mylabel: myvalue
+    # ...
+----
+
+.Supported resource in Kafka cluster
+
+The `template` object for the Kafka cluster can contain following fields:
+
+`statefulset`:: Configures the Kafka broker `StatefulSet`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the StatefulSet used by the Kafka broker.

File: documentation/book/con-customizing-template-properties.adoc [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+`pod`:: Configures the Kafka broker `Pods` created by the `StatefulSet`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Please remove monospace from Pods and StatefulSet.

File: documentation/book/con-customizing-template-properties.adoc [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+`bootstrapService`:: Configures the bootstrap service used for access by clients running inside {ProductPlatformName}.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the bootstrap service used by clients running within {ProductPlatformName} to connect to the Kafka broker.

File: documentation/book/con-customizing-template-properties.adoc [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+`brokersService`:: Configures the headless service.
+`externalBootstrapService`:: Configures the bootstrap service used by clients connecting to Kafka brokers from outside of {ProductPlatformName}.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the bootstrap service used by clients connecting to the Kafka broker from outside {ProductPlatformName}.

File: documentation/book/con-customizing-template-properties.adoc [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+`perPodService`:: Configures the services used by clients connecting to Kafka brokers from outside of {ProductPlatformName} to access the individual brokers.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the per-Pod services used by clients connecting to the Kafka broker from outside {ProductPlatformName} to access individual brokers.

File: documentation/book/con-customizing-template-properties.adoc [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+`externalBootstrapRoute`:: Configures the bootstrap route used by clients connecting to Kafka brokers from outside of {OpenShiftName} using {OpenShiftName} `Routes`.
+`perPodRoute`:: Configures the routes used by clients connecting to Kafka brokers from outside of {OpenShiftName} using {OpenShiftName} `Routes` to access the individual brokers.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the per-Pod routes used by clients connecting to the Kafka broker from outside {OpenShiftName} to access individual brokers.

File: documentation/book/con-customizing-template-properties.adoc [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+.Supported resource in Zookeeper cluster
+
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: _Add an introductory sentence:_

When defined in a Zookeeper cluster, the `template` object can have the following fields:

File: documentation/book/con-customizing-template-properties.adoc [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+`statefulset`:: Configures the Zookeeper `StatefulSet`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the StatefulSet used by the Zookeeper.

File: documentation/book/con-customizing-template-properties.adoc [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+`pod`:: Configures the Zookeeper `Pods` created by the `StatefulSet`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: _Please remove the monospace from Pods and StatefulSet._

File: documentation/book/con-customizing-template-properties.adoc [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+`clientsService`:: Configures the service used by clients to access Zookeeper.
+`nodesService`:: Configures the headless service.
+
+.Supported resource in Entity Operator
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: When defined in an Entity Operator , the `template` object can have the following fields:

File: documentation/book/con-customizing-template-properties.adoc [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+`deployment`:: Configures the Entity Operator `Deployment`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Configures the Deployment used by the Entity Operator.

File: documentation/book/con-customizing-template-properties.adoc [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+`pod`:: Configures the Entity Operator `Pod` created by the `Deployment`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: _Remove monospace from Pod and Deployment_

File: documentation/book/con-customizing-template-properties.adoc [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+
+.Supported resource in Kafka Connect and Kafka Connect with Source2Image support
+
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: When used with Kafka Connect and Kafka Connect with Source2Image support , the template object can have the following fields:

File: documentation/book/con-customizing-template-properties.adoc [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+`deployment`:: Configures the Kafka Connect `Deployment`.
+`pod`:: Configures the Kafka Connect `Pods` created by the `Deployment`.
+`apiService`:: Configures the service used by Kafka Connect REST API.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: Kafka Connect REST API > the Kafka Connect REST API

File: documentation/book/con-customizing-template-properties.adoc [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+`nodesService`:: Configures the headless service.
+
+.Supported resource in Entity Operator
+
+`deployment`:: Configures the Entity Operator `Deployment`.
+`pod`:: Configures the Entity Operator `Pod` created by the `Deployment`.
+
+.Supported resource in Kafka Connect and Kafka Connect with Source2Image support
+
+`deployment`:: Configures the Kafka Connect `Deployment`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: _Please remove monospace from Deployment and Pods._

File: documentation/book/con-customizing-template-properties.adoc [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+`pod`:: Configures the Kafka Connect `Pods` created by the `Deployment`.
+`apiService`:: Configures the service used by Kafka Connect REST API.
+
+.Supported resource in Kafka Mirror Maker
+
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: When used with Kafka Mirror Maker , the `template` object can have the following fields:

File: documentation/book/con-customizing-template-properties.adoc [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+`deployment`:: Configures the Kafka Mirror Maker `Deployment`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-template-properties.adoc]: _Remove monospace as above._

File: documentation/book/con-customizing-labels-and-annotations.adoc [hunk 27]
```diff
@@ -0,0 +1,28 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-labels-and-annotations-{context}']
+= Labels and Annotations
+
+For every resource, you can configure additional `Labels` and `Annotations`.
+The `Labels` and `Annotations` are configured in the `metadata` property.
```
Comment (laidan6000) [inline documentation/book/con-customizing-labels-and-annotations.adoc]: The `Labels` and `Annotations` > `Labels` and `Annotations`

File: documentation/book/con-customizing-labels-and-annotations.adoc [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+            labels:
+                label1: value1
+                label2: value2
+            annotations:
+                annotation1: value1
+                annotation2: value2
+# ...
+----
+
+The `labels` and `annotations` fields can contain any labels or annotations which do not contain the reserved string `strimzi.io`.
```
Comment (laidan6000) [inline documentation/book/con-customizing-labels-and-annotations.adoc]: which do not contain > that do not contain

File: documentation/book/proc-customizing-deployments.adoc [hunk 29]
```diff
@@ -0,0 +1,56 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='proc-customizing-deployments-{context}']
+= Customizing deployments
+
```
Comment (laidan6000) [inline documentation/book/proc-customizing-deployments.adoc]: _Add an introduction:_

This procedure describes how to customize a Kaka deployment using `Labels`. You can also use `Annotations`.

File: documentation/book/proc-customizing-deployments.adoc [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+.Prerequisites
+
+* An {ProductPlatformName} cluster
```
Comment (laidan6000) [inline documentation/book/proc-customizing-deployments.adoc]: Add full stops to end of bullet points.

File: documentation/book/proc-customizing-deployments.adoc [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+On {KubernetesName} this can be done using `kubectl apply`:
+[source,shell,subs=+quotes]
+kubectl apply -f _your-file_
++
+Or using `kubectl edit`:
+[source,shell,subs=+quotes]
+kubectl edit _Resource_ _ClusterName_
++
+endif::Kubernetes[]
+On {OpenShiftName} this can be done using `oc apply`:
```
Comment (laidan6000) [inline documentation/book/proc-customizing-deployments.adoc]: On {OpenShiftName}, use `oc apply`:

File: documentation/book/proc-customizing-deployments.adoc [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
++
+[source,shell,subs=+quotes]
+oc apply -f _your-file_
++
+Or using `oc edit`:
```
Comment (laidan6000) [inline documentation/book/proc-customizing-deployments.adoc]: Alternatively, use `oc edit`:

**Pull request merged**